libbeat/processors/add_cloud_metadata/provider_aws_ec2_test.go

```diff
@@ -75,23 +75,49 @@ func TestRetrieveAWSMetadata(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	actual, err := p.Run(&beat.Event{Fields: common.MapStr{}})
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	expected := common.MapStr{
-		"cloud": common.MapStr{
-			"provider": "ec2",
-			"instance": common.MapStr{
-				"id": "i-11111111",
+	cases := []struct {
+		fields          common.MapStr
+		expectedResults common.MapStr
+	}{
+		{
+			common.MapStr{},
+			common.MapStr{
+				"cloud": common.MapStr{
```

Comment:
Could you also add a test where we have `cloud.provider: ecs2` to see what happens?

Reply:
Good point, I added the test and turned out `"cloud": common.MapStr{"provider": "ec2"}` works but `"cloud.provider": "ec2"` does not work. Function `GetValue` does not think `cloud.provider` has a `cloud` key.